### PR TITLE
Add more keywords as identifier in PPL

### PR DESCRIPTION
**Pull request by penghuo**
Signed-off-by: Peng Huo <penghuo@gmail.com>

Solve #1318
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
Can you copy-paste this for SQL?

**Comment by penghuo** [comment]
> Can you copy-paste this for SQL?

This is quick fix for PPL which is required by Observability. 
https://github.com/opensearch-project/sql/pull/1191 already cover SQL releated identifier change.

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 1]
```diff
@@ -614,8 +614,26 @@ keywordsCanBeId
...
+    | SORTBY
+    // FIELDKEYWORDSAUTO
+    | STR | IP | NUM
+    // ARGUMENT KEYWORDS
+    | KEEPEMPTY | CONSECUTIVE | DEDUP_SPLITVALUES | PARTITIONS | ALLNUM | DELIM | CENTROIDS | ITERATIONS | DISTANCE_TYPE
+    | NUMBER_OF_TREES | SHINGLE_SIZE | SAMPLE_SIZE | OUTPUT_AFTER | TIME_DECAY | ANOMALY_RATE | CATEGORY_FIELD
+    | TIME_FIELD | TIME_ZONE | TRAINING_DATA_SIZE | ANOMALY_SCORE_THRESHOLD
+    // AGGREGATIONS
+    | AVG | COUNT | DISTINCT_COUNT | ESTDC | ESTDC_ERROR | MAX | MEAN | MEDIAN | MIN | MODE | RANGE | STDEV | STDEVP
+    | SUM | SUMSQ | VAR_SAMP | VAR_POP | STDDEV_SAMP | STDDEV_POP | PERCENTILE | TAKE | FIRST | LAST | LIST | VALUES
```
[the remark this answers is not in the document]
Comment (penghuo) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:632]: done.

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 2]
```diff
@@ -614,8 +614,26 @@ keywordsCanBeId
     | TIMESTAMP | DATE | TIME
```
[the remark this answers is not in the document]
Comment (penghuo) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4]: done.

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by anirudha — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**